ROX-13396: Add SAML authentication method for AWS (#544)
stackrox/acs-fleet-manager: docs/development/secret-management.md

- acs-prod-dp-01

## Instructions

> - `AWS_AUTH_HELPER` environment variable is responsible for choosing the appropriate authentication method within the deployment scripts. Possible options are:

Reviewer: ```suggestion
- `AWS_AUTH_HELPER` environment variable selects the appropriate authentication method within the deployment scripts. Possible options are:
```

Author: Dome

  - `aws-vault`
  - `aws-saml`
  - `none` (default)
- Depending on the environment, the following choices are set:

    | Source | Target      | AWS_AUTH_HELPER |
    |--------|-------------|-----------------|
    | local  | dev         | aws-vault       |
    | local  | stage,prod  | aws-saml        |
    | CI/CD  | stage,prod  | none            |

> - For SAML authentication, you must have access to the appropriate `aws-automation` git repository to download the tool.

Reviewer: ```suggestion
- For SAML authentication, you must have access to the `aws-automation` git repository to download the tool.
```
Is it behind VPN? If so it would be worth mentioning here.

Author: Done
